src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs

```diff
@@ -32,6 +32,14 @@ public class StoreDatasetValidator : IStoreDatasetValidator
     private readonly StoreMeter _storeMeter;
     private readonly IDicomRequestContextAccessor _dicomRequestContextAccessor;
 
+    private static readonly HashSet<DicomTag> RequiredCoreTags = new HashSet<DicomTag>()
+    {
+        DicomTag.StudyInstanceUID,
+        DicomTag.SeriesInstanceUID,
+        DicomTag.SOPInstanceUID,
+        DicomTag.PatientID,
```

Comment:
We'll want to add `SOPClassUID` to this list. Need an action item on deciding whether it should be required in V2 (isn't indexed).